Author: Jan Christoph Ebersbach <jceb@e-jc.de>
URL: http://dwm.suckless.org/patches/clear-layout
clear layout hides all visible clients so that you can enjoy your nice wallpaper

Index: dwm-patches/dwm/clear.c
===================================================================
--- /dev/null
+++ dwm-patches/dwm/clear.c
@@ -0,0 +1,9 @@
+void
+clear(Monitor *m) {
+	Client *c;
+	for (c = m->stack; c; c = c->snext) {
+		if(ISVISIBLE(c)) {
+			XMoveWindow(dpy, c->win, WIDTH(c) * -2, c->y);
+		}
+	}
+}
Index: dwm-patches/dwm/config.def.h
===================================================================
--- dwm-patches.orig/dwm/config.def.h
+++ dwm-patches/dwm/config.def.h
@@ -64,6 +64,7 @@ static const Layout layouts[] = {
 	{ "+++",      gaplessgrid },
 	{ "|M|",      centeredmaster },
 	{ ">M>",      centeredfloatingmaster },
+	{ "[ ]",      clear },
 };
 
 /* key definitions */
@@ -92,6 +93,7 @@ static const char *termcmd[]  = { "x-ter
 #include "focusmaster.c"
 #include "shifttags.c"
 #include "focusurgent.c"
+#include "clear.c"
 static Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
@@ -115,6 +117,7 @@ static Key keys[] = {
 	{ MODKEY,                       XK_g,      setlayout,      {.v = &layouts[3]} },
 	{ MODKEY,                       XK_u,      setlayout,      {.v = &layouts[4]} },
 	{ MODKEY,                       XK_y,      setlayout,      {.v = &layouts[5]} },
+	{ MODKEY,                       XK_c,      setlayout,      {.v = &layouts[6]} },
 	{ MODKEY,                       XK_space,  toggleview_focus, {.ui = 1} },
 	{ MODKEY|ControlMask,           XK_space,  setlayout,      {0} },
 	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
Index: dwm-patches/dwm/dwm.c
===================================================================
--- dwm-patches.orig/dwm/dwm.c
+++ dwm-patches/dwm/dwm.c
@@ -158,6 +158,7 @@ static void centeredfloatingmaster(Monit
 static void checkotherwm(void);
 static void cleanup(void);
 static void cleanupmon(Monitor *mon);
+static void clear(Monitor *m);
 static void clientmessage(XEvent *e);
 static void configure(Client *c);
 static void configurenotify(XEvent *e);
